### PR TITLE
start prep for v1.2 upgrade instructions

### PR DESCRIPTION
**Pull request by malaffoon**
Mostly this was to capture the changes made to config properties. But i set up the upgrade instructions so we have a place to record changes as we make them.

### CODE REVIEW

File: deploy/Upgrade.v1_2.AWS.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+* Configurable Subjects
+* Configure Required/Optional Fields for Ingest Validation
+* Teacher-Created Groups
+* Unknown Gender
+* English Language Acquisition Status
+
+It involves some minor architectural changes:
+* Reorganize application and tenant properties
+
+TODO - Because there are schema changes, the data warehouse and reporting data mart must be migrated. This may take a couple hours depending on the amount of data involved. Combined with the non-trivial changes to the system configuration, this means the upgrade process may take 2-4 hours. It is important to alert the user base and any 3rd party data feeds of the extended downtime required for this upgrade.
```
Comment (malaffoon) [inline deploy/Upgrade.v1_2.AWS.md]: this may not apply, don't really know yet, but left it in for now

Comment (agorina) [reply deploy/Upgrade.v1_2.AWS.md]: One thing that we know is that because of IABs we will need to clean and reload Redshift. This takes close to an hour.

Comment (agorina) [reply deploy/Upgrade.v1_2.AWS.md]: I also suggest to add to the instructions to run validation scripts before and after the update.

Comment (malaffoon) [reply deploy/Upgrade.v1_2.AWS.md]: done

**Pull request merged**